File: app/src/jvx/java/art/coded/wireframe/view/adapter/ListAdapter.java
package art.coded.wireframe.view.adapter;

import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.app.ShareCompat;
import androidx.recyclerview.widget.RecyclerView;

import art.coded.wireframe.R;
import art.coded.wireframe.model.entity.Element;

import java.util.List;

public class ListAdapter extends RecyclerView.Adapter<ListAdapter.ElementViewHolder> {

    private static final String LOG_TAG = ListAdapter.class.getSimpleName();

    LayoutInflater mLayoutInflater;
    List<Element> mAllElements;
    Activity mActivity;

    public ListAdapter(Activity activity) {

        mActivity = activity;
        mLayoutInflater = LayoutInflater.from(activity);
    }

    @NonNull @Override  public ElementViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View itemView = mLayoutInflater.inflate(R.layout.list_item, parent, false);
        return new ElementViewHolder(itemView, mActivity);
    }

    @Override  public void onBindViewHolder(@NonNull ElementViewHolder holder, int position) {

        if (mAllElements != null) {

            Element element = mAllElements.get(position);
            holder.mItemView.setText(element == null ? "No elements" : element.getName());
        }
    }

    @Override  public int getItemCount() {

        return mAllElements == null ? 0 : mAllElements.size();
    }

    public void setElements(List<Element> elements) {
        mAllElements = elements;
        notifyDataSetChanged();
    }

    public Element getNameByPosition(int position) { return mAllElements.get(position); }

    static class ElementViewHolder extends RecyclerView.ViewHolder {

        private final TextView mItemView;
        private final ImageButton mImageButton;

        public ElementViewHolder(@NonNull View itemView, Activity activity) {

            super(itemView);
            mItemView = itemView.findViewById(R.id.textView);
            mImageButton = itemView.findViewById(R.id.list_button_share);
            mImageButton.setOnClickListener(view -> {

                ShareCompat.IntentBuilder
                        .from(activity)
                        .setType("text/plain")
                        .setText(mItemView.getText())
                        .startChooser();
            });
        }
    }
}

File: app/src/main/java/art/coded/wireframe/view/adapter/ListAdapter.kt
package art.coded.wireframe.view.adapter

import android.app.Activity
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import art.coded.wireframe.R
import android.widget.TextView
import android.widget.ImageButton
import androidx.core.app.ShareCompat
import android.view.View
import art.coded.wireframe.model.entity.Element

class ListAdapter(var mActivity: Activity?) :
    RecyclerView.Adapter<ListAdapter.ElementViewHolder>() {
    var mLayoutInflater: LayoutInflater
    var mAllElements: List<Element?>? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ElementViewHolder {
        val itemView = mLayoutInflater.inflate(R.layout.list_item, parent, false)
        return ElementViewHolder(itemView, mActivity)
    }

    override fun onBindViewHolder(holder: ElementViewHolder, position: Int) {
        if (mAllElements != null) {
            val element = mAllElements!![position]
            holder.mItemView.text = element?.name ?: "No elements"
        }
    }

    override fun getItemCount(): Int {
        return if (mAllElements == null) 0 else mAllElements!!.size
    }

    fun setElements(elements: List<Element?>?) {
        mAllElements = elements
        notifyDataSetChanged()
    }

    fun getNameByPosition(position: Int): Element? {
        return mAllElements!![position]
    }

    class ElementViewHolder(itemView: View, activity: Activity?) :
        RecyclerView.ViewHolder(itemView) {
        val mItemView: TextView
        private val mImageButton: ImageButton

        init {
            mItemView = itemView.findViewById(R.id.textView)
            mImageButton = itemView.findViewById(R.id.list_button_share)
            mImageButton.setOnClickListener { view: View? ->
                ShareCompat.IntentBuilder
                    .from(activity!!)
                    .setType("text/plain")
                    .setText(mItemView.text)
                    .startChooser()
            }
        }
    }

    companion object {
        val LOG_TAG = ListAdapter::class.java.simpleName
    }

    init {
        mLayoutInflater = LayoutInflater.from(mActivity)
    }
}